crates/telio-batcher/src/batcher.rs

```diff
@@ -99,11 +132,25 @@ where
         action: Action<V>,
     ) {
         telio_log_debug!(
-            "adding item to batcher with key({:?}), interval({:?}), threshold({:?})",
+            "Adding item to batcher with key({:?}), interval({:?}), threshold({:?})",
             key,
             interval,
             threshold,
         );
+
+        let threshold = {
+            if threshold >= interval {
+                let capped_threshold = interval / 2;
```

Comment:
If the interval is greater than the threshold, why is it capped at half the interval, and not lets say any other value (like 1/4 interval for eg). Just want to understand the reasoning here.

Reply:
No worries, could have commented that, but no real reason except that based on our design sessions and RFCs it seems that the threshold of half the period should work the best as it should batch everything eventually that way but also is not too big to needlessly send data (batcher is currently replacement for wireguard's persistent-keepalives)

It would probably even be better if this check would be more verbose, since if difference is miniscule but still not enough to pass the conditional
it will enter a spiral where:
- A sends data to B
- B get's triggered, sends what's batched to A
- A get's trigger in turn, so sends it's batched things to A
- repeat

Reply:
@LukasPukenis note to self: introduce a minimum threshold, such as 5seconds to avoid spiral of death. However even 5 seconds might be too much if the interval is for example 3 seconds, so the check should be for a difference:

```
if interval - threshold < HARD_THRESHOLD_LIMIT {
  threshold = floor(interval / 2); // as per rfc this value is our guess to be the best value, so we default to that
}
```

Reply:
@Hasan6979 your question triggered me(pun not intended) to work a bit more on it and I've added minimal checks into the code as well:
- each item has a timestamp it last emitted action which is then checked against
- threshold is capped at maximum of half the interval instead of the previous check with the reason being that it's much more secure that way and easier to verify the correct behaviour.